.github/workflows/test.yaml

```diff
@@ -32,23 +25,31 @@ jobs:
         python-version: ${{ fromJson(needs.get-python-versions.outputs.python-matrix) }}
 
     steps:
-      - uses: actions/checkout@v3
-        with:
-          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Check out repository
+        uses: actions/checkout@v3
 
       - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+        uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: Load cached Poetry installation
+        uses: actions/cache@v3
+        with:
+          path: ~/.local # the path depends on the OS
+          key: poetry-0 # increment to reset cache
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+
+      - name: Specify Poetry Python version
+        run: poetry env use ${{ matrix.python-version }}
+
```

Comment:
There's been a lot of kerfuffle about this and not sure how much it matters because the build was super fast but in the check it looks like it hit the cache but then still installed the dependencies again?

Reply:
nice!